### ISSUE TITLE
video calling enhancement!

### ISSUE THREAD

**Issue by Utkarsh-Pandey-GitHub** [Yeasir0032/Discord-Clone#31]
**feature 

video channels and audio channels can be combined and enhanced.

**Describe the solution you'd like**
I recently made a video call miniproject and here are some changes i would like to add
- [ ] 1. a waiting room or a place to go back after you leave the meet
![image](https://github.com/Yeasir0032/Discord-Clone/assets/92950393/a20498f0-5472-45bf-8757-6a356007a644)
- [ ] 2. working screenshare and a single video conferencing channel
![image](https://github.com/Yeasir0032/Discord-Clone/assets/92950393/741d3bea-632f-4476-bac3-1e1efe037a51)
- [ ] 3. [optional] recording of videos

**Comment by Yeasir0032** [comment]
Did you test the app?
These features already exist. Do you want to enhance these features?

**Comment by Yeasir0032** [comment]
It uses Livekit for video audio conferencing. 
It will be great if it is replaced and built from scratch.

**Comment by Utkarsh-Pandey-GitHub** [comment]
> Did you test the app? These features already exist. Do you want to enhance these features?



> It uses Livekit for video audio conferencing. It will be great if it is replaced and built from scratch.

yes i have tested these and I build that app with stream.io. it is similar to live kit but better . 
currently in this app the video ,audio and screenshare are not working properly but with stream.io we can provide many more feature.
my demo app: https://wknote.vercel.app/videoCall
stream.io documentation: https://getstream.io/video/docs/react/

**Comment by Yeasir0032** [comment]
With Livekit everything is working fine isn't it? Or did you find a bug?

Livekit and streamio are quite same of a kind. Can you create it from scratch (handling it's backend) ?

**Comment by mahek0620** [comment]
@Yeasir0032 can you assign this task to me
